sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -183,31 +206,108 @@ class CompactBlockProcessor internal constructor(
         // Clear any undeleted left over block files from previous sync attempts
         deleteAllBlockFiles(
             downloader = downloader,
-            lastKnownHeight = getLastScannedHeight(repository)
+            lastKnownHeight = when (val result = getMaxScannedHeight(backend)) {
+                is GetMaxScannedHeightResult.Success -> result.height
+                else -> null
+            }
```

Comment:
This looks incorrect to me; given out-of-order sync, it could delete block files that have been downloaded but not yet scanned because the max-scanned height may be near the tip of the chain, but un-scanned files can exist below this height. Use `getFullyScannedHeight` instead.

Reply:
Oh, actually, I didn't realize that `lastKnownHeight` is just used for an error message. Never mind.

Reply:
Yes, it only deletes persisted leftover block files from the disc, and the `lastKnownHeight` is for unified error reporting across sync substages.